OpenRA.Mods.Common/Widgets/Logic/SettingsLogic.cs

```diff
@@ -190,14 +190,21 @@ Action InitDisplayPanel(Widget panel)
 
 			var frameLimitTextfield = panel.Get<TextFieldWidget>("FRAME_LIMIT_TEXTFIELD");
 			frameLimitTextfield.Text = ds.MaxFramerate.ToString();
-			frameLimitTextfield.OnLoseFocus = () =>
+			frameLimitTextfield.OnEnterKey = () =>
 			{
 				int fps;
 				Exts.TryParseIntegerInvariant(frameLimitTextfield.Text, out fps);
 				ds.MaxFramerate = fps.Clamp(1, 1000);
 				frameLimitTextfield.Text = ds.MaxFramerate.ToString();
+				frameLimitTextfield.YieldKeyboardFocus();
+				return true;
+			};
+			frameLimitTextfield.OnEscKey = () =>
```

Comment:
And in the lobby, if you alternatingly click on chat field and player name field, you are also adding them multiple times to the stack. But there it doesn't cause any problems because chat field doesn't yield focus when ESC is pressed, it just resets the text, so there is no focus switching with ESC. And when the lobby widget is removed, these text fields become null and stay on the stack and get cleaned up with the next YieldKeyboardFocus() call.